src/classes/TERM_CannotDelete_TEST.cls

```diff
@@ -39,237 +39,282 @@ private class TERM_CannotDelete_TEST {
         termGradesToInsert.add(termGrade);
-        
+
         // Insert Application for Term 3
         List<Application__c> applicaitonsToInsert = new List<Application__c>();
         Application__c application =  UTIL_UnitTestData_TEST.getApplication(academicAccount.Id, student.Id, term3.Id);
         applicaitonsToInsert.add(application);
-        
+
         // Insert Course offering, Term Grades and Application for Term 4
         Course_Offering__c courseOffering2 = UTIL_UnitTestData_TEST.createCourseOffering(NULL, term4.Id);
-        
+
         Term_Grade__c termGrade2 = UTIL_UnitTestData_TEST.getTermGradeWTermCourseConn(term4.Id, courseConnection.Id);
         termGradesToInsert.add(termGrade2);
-        
+
         // Insert Term Grades
         insert termGradesToInsert;
-        
+
         Application__c application2 =  UTIL_UnitTestData_TEST.getApplication(academicAccount.Id, student.Id, term4.Id);
         applicaitonsToInsert.add(application2);
-        
+
         // Insert Applicaitons
         insert applicaitonsToInsert;
-        
+
         Test.startTest();
             List<Term__c> termsToDelete = [SELECT Id FROM TERM__c];
-            Database.DeleteResult[] results = Database.delete(termsToDelete, false);
+            Database.DeleteResult[] results = Database.delete(termsToDelete, FALSE);
         Test.stopTest();
-        
+
         // Ensure all Terms are deleted
         List<Term__c> termsAfterDelete = [SELECT Id FROM TERM__c];
-        System.assertEquals(0, termsAfterDelete.size());       
-    }   
+        System.assertEquals(0, termsAfterDelete.size());
+    }
+
+    /*********************************************************************************************************
+    * @description Tests the deletion of Term with Term as child record
+    */
+    @isTest
+    static void testDeleteTermsWithTerms() {
+        
+        // Enable Prevent Term Deletion Cusotm Settings
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Prevent_Term_Deletion__c = TRUE));
+
+        // Insert Account
+        List<Account> academicAccount = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, UTIL_Describe_API.getAcademicAccRecTypeID());
+        insert academicAccount;
+
+        Term__c term1 = [SELECT Id, Parent_Term__c FROM TERM__c];
+
+        // Insert Term2 with Term1 as the Parent Term
+        Term__c term2 = UTIL_UnitTestData_TEST.getTerm(academicAccount[0].Id, 'Fall 19');
+        term2.Parent_Term__c = term1.Id;
+        insert term2;
+       
+       // Ensure there are 2 terms
+        List<Term__c> termsBeforeDelete = [SELECT Id FROM TERM__c];
+        System.assertEquals(2, termsBeforeDelete.size());
+
+        // Ensure the Parent Term is not deleted
+        try {
+            Test.startTest();
+                List<Term__c> termsToDelete = [SELECT Id FROM TERM__c ORDER BY Name ASC];
```

Comment:
@skosgi1  -- Let's check and see if we can actually do ORDER BY Name here without breaking shield. I don't believe we can.

Reply:
looks like we cannot: https://help.salesforce.com/articleView?id=security_pe_considerations_general.htm&type=5

Reply:
Fixed in the latest commit. Resolving convo.